Tests/UnitTests/Core/Vertexing/LinearizedTrackFactoryTests.cpp

```diff
@@ -220,41 +153,116 @@ BOOST_AUTO_TEST_CASE(linearized_track_factory_straightline_test) {
     // Fill vector of track objects with simple covariance matrix
     Covariance covMat;
 
+    // TODO: add time
     covMat << resD0 * resD0, 0., 0., 0., 0., 0., 0., resZ0 * resZ0, 0., 0., 0.,
         0., 0., 0., resPh * resPh, 0., 0., 0., 0., 0., 0., resTh * resTh, 0.,
         0., 0., 0., 0., 0., resQp * resQp, 0., 0., 0., 0., 0., 0., 1.;
     tracks.emplace_back(perigeeSurface, paramVec, std::move(covMat));
   }
 
-  // Set up helical track linearizer for the case of a non-existing
-  // magnetic field, which results in the extreme case of a straight line
-  LinearizerStraightLine::Config ltConfig(propagator);
-  LinearizerStraightLine linFactory(ltConfig);
-
-  auto field = std::make_shared<NullBField>();
-
-  LinearizerStraightLine::State state(field->makeCache(magFieldContext));
-
-  BoundVector vecBoundZero = BoundVector::Zero();
-  BoundSymMatrix matBoundZero = BoundSymMatrix::Zero();
-  Vector4 vecSPZero = Vector4::Zero();
-  ActsMatrix<eBoundSize, 4> matBound2SPZero = ActsMatrix<eBoundSize, 4>::Zero();
-  ActsMatrix<eBoundSize, 3> matBound2MomZero =
-      ActsMatrix<eBoundSize, 3>::Zero();
-
-  for (const BoundTrackParameters& parameters : tracks) {
-    LinearizedTrack linTrack =
-        linFactory
-            .linearizeTrack(parameters, Vector4::Zero(), geoContext,
-                            magFieldContext, state)
-            .value();
-
-    BOOST_CHECK_NE(linTrack.parametersAtPCA, vecBoundZero);
-    BOOST_CHECK_NE(linTrack.covarianceAtPCA, matBoundZero);
-    BOOST_CHECK_EQUAL(linTrack.linearizationPoint, vecSPZero);
-    BOOST_CHECK_NE(linTrack.positionJacobian, matBound2SPZero);
-    BOOST_CHECK_NE(linTrack.momentumJacobian, matBound2MomZero);
-    BOOST_CHECK_NE(linTrack.constantTerm, vecBoundZero);
+  // Linearizer for constant field and corresponding state
+  AnalyticalLinearizer::Config linConfig(constField, propagator);
+  AnalyticalLinearizer linFactory(linConfig);
+  AnalyticalLinearizer::State linState(constField->makeCache(magFieldContext));
+
+  NumericalLinearizer::Config numLinConfig(constField, propagator);
+  NumericalLinearizer numLinFactory(numLinConfig);
+  NumericalLinearizer::State numLinState(
+      constField->makeCache(magFieldContext));
```

Comment:
For constant and zero fields, the cache has infinite validity, to technically you could reuse it. It won't really make a difference here, just mentioning it.